docs/DeveloperGuide.md

```diff
@@ -158,21 +158,51 @@ Classes used by multiple components are in the `seedu.address.commons` package.
 
 This section describes some noteworthy details on how certain features are implemented.
 
+### Lesson Time Parameter
+This parameter allows users to keep track of a student's lesson timings. Multiple lesson times can be added for a single student.
+
+#### Design Considerations
+**Treatment of clashing timings**
+
+- **Current Implementation (Alternative 1):**
+    - **Description**: Clashing lesson timings across and within students are not rejected.
+    - **Pros**: Promotes greater user flexibility, as there could be different schedules for alternate weeks,
+      and all the timings should be stored in EduManage.
+    - **Cons**: Wrong lesson timings would not be flagged out to the user.
+
+- **Alternative 2:**
+    - **Description**: Reject clashing lesson timings across and within students.
+    - **Pros**: Flags out wrong lesson timings input by the user.
+    - **Cons**: Reduces user flexibility in storing lesson times.
+
+**Association between lesson timings and subjects**
+
+- **Current Implementation (Alternative 1):**
+    - **Description**: Each lesson timing is not associated with any subject.
+    - **Pros**: Promotes greater user flexibility, as users might be asked to teach different subjects within
+      a lesson timing.
+    - **Cons**: Users might prefer to have the association made clear.
+
+- **Alternative 2:**
+    - **Description**: Associate each lesson timing with a specific subject.
+    - **Pros**: Users can clearly see what they are supposed to teach and when.
```

Comment:
Clarifies what subject is to be taught during each lesson, helping users stay organized.